app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -80,6 +82,33 @@ class HomeFragment : Fragment() {
             }
             override fun onTransitionCompleted(p0: MotionLayout?, p1: Int) { }
         })
+
+        privateBrowsingButton.setOnClickListener {
+            // Get the current private mode preference and flip it
+            PreferenceManager.getDefaultSharedPreferences(context!!)
+                .edit()
+                .putBoolean(getString(R.string.pref_key_private_mode),
+                    !PreferenceManager.getDefaultSharedPreferences(context!!)
+                    .getBoolean(getString(R.string.pref_key_private_mode), false))
+                .apply()
+
+            applyPrivateModeIfNecessary()
+        }
+    }
+
+    private fun applyPrivateModeIfNecessary() {
+        val isPrivate = PreferenceManager.getDefaultSharedPreferences(context!!)
```

Comment:
On the same vein, this logic can be pulled into the theme coordinator/provider. And the HomeActivity can potentially observe changes of this model and re-create itself. Versus having all of this logic in the HomeFragment.